### PR TITLE
test: Add manual scenario for network polling scenario

### LINKED ISSUE #25964: Add manual scenario to ensure that network polling stops when UI is closed
### What is this about?

We have release blocker there is periodic polling task running every 20 seconds even when UI is closed), it generates 1 API request to Etherscan and 3 to RPC provider:
API request to Etherscan:
` https://api.etherscan.io/api?module=account&address=0x5197b5b062288bbf29008c92b08010a92dd677cd&startBlock=20327381&offset=40&sort=desc&action=txlist&tag=latest&page=1`
3 to RPC provider:
```
"jsonrpc":"2.0","method":"eth_blockNumber",
"jsonrpc":"2.0","method":"eth_getBlockByNumber"
jsonrpc":"2.0","method":"eth_call","params":
```
This ticket aims to add a manual scenario to prevent it happen again.

Slack discussion: https://consensys.slack.com/archives/C07CGH75GFN/p1721233816417369
Related PR attempt to fix: https://github.com/MetaMask/metamask-extension/pull/25909

To repro the bug:

* Install MM
* Open the service worker dev console
* Open and close the pop up
* Go to the network tab of the dev console and see that requests, continue to occur ever ~20 seconds.

### Scenario

_No response_

### Design

_No response_

### Technical Details

_No response_

### Threat Modeling Framework

_No response_

### Acceptance Criteria

_No response_

### Stakeholder review needed before the work gets merged

- [ ] Engineering (needed in most cases)
- [ ] Design
- [ ] Product
- [ ] QA (automation tests are required to pass before merging PRs but not all changes are covered by automation tests - please review if QA is needed beyond automation tests)
- [ ] Security
- [ ] Legal
- [ ] Marketing
- [ ] Management (please specify)
- [ ] Other (please specify)

### References

_No response_

### PR DESCRIPTION
**Pull request by chloeYue**
## **Description**

This PR introduces a new test scenario which relates to a release-blocker for v12.0.0. This scenario is to verify that once the MetaMask UI is closed, the extension stops all network requests, including `etherscan`, `eth_call`, `eth_getBalance`, `eth_blockNumber`, and `eth_getBlockByNumber` to Infura or any other RPC provider.  This check is vital for reducing unnecessary network traffic and enhancing both privacy and efficiency within the extension's operation.

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/26195?quickstart=1)

## **Related issues**

Fixes: #25964
Related release blocker: #25951 

## **Manual testing steps**
Follow testing steps in this  manual scenario

## **Pre-merge author checklist**

- [x] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [x] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [x] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: test/manual-scenarios/security/stop-network-polling-scenario.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+- **Given** the user has installed the MetaMask extension
+- **And** the user has onboarded successfully
+
+### Scenario: Verify no network requests are made after MetaMask UI is closed
+
+1. **Given** the user opens the MetaMask extension
+2. **And** the user navigates to the service worker dev console's network tab
+3. **When** the user observes "etherscan", "eth_call", "th_blockNumber" and "eth_getBlockByNumber" network requests
+4. **And** then closes the MetaMask UI
```
[the remark this answers is not in the document]
Comment (chloeYue) [reply test/manual-scenarios/security/stop-network-polling-scenario.md]: Good catch, thanks !

File: test/manual-scenarios/security/stop-network-polling-scenario.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
+# Stop Network Polling When MetaMask UI Is Closed
+
+## Feature Description
+
+This test scenario ensures that after closing the MetaMask UI, the extension stops sending network requests, including `etherscan`, `eth_call`, `eth_blockNumber`, and `eth_getBlockByNumber`, to Infura or any other RPC provider. This behavior is crucial for preventing unnecessary network traffic and ensuring privacy and efficiency.
```
Comment (seaona) [inline test/manual-scenarios/security/stop-network-polling-scenario.md:5]: the account polling also throw this one: `eth_getBalance` , maybe we could also include it explicitly here?

see: https://github.com/MetaMask/metamask-extension/pull/26141/files#diff-174d8183603b0004a7ff4acb5a21374ea2ad88e63ab433090783ef41db9c35e7R153

**Review by seaona — DISMISSED** [review]
LGTM, just a small suggestion but non-blocking

File: test/manual-scenarios/security/stop-network-polling-scenario.md (hunk 2 above)
Comment (chloeYue) [reply test/manual-scenarios/security/stop-network-polling-scenario.md]: make sense, added in this commit [a754fb5](https://github.com/MetaMask/metamask-extension/pull/26195/commits/a754fb59c475e6c9b52d56079e99e208c2024bfd) thanks for the nice catch !

**Comment by chloeYue** [comment]
> LGTM, just a small suggestion but non-blocking

Nice catch, added in [a754fb5](https://github.com/MetaMask/metamask-extension/pull/26195/commits/a754fb59c475e6c9b52d56079e99e208c2024bfd), thanks !

**Review by seaona — APPROVED** [review]

**Review by DDDDDanica — APPROVED** [review]

**Pull request merged**